Microsoft.Toolkit.Uwp.UI.Animations/Expressions/ExpressionNodes/ExpressionNode.cs

```diff
@@ -312,6 +313,22 @@ internal void EnsureReferenceInfo()
                     refNode.ParamName = paramName;
                 }
             }
+
+            // Generates Excel-column-like identifiers, e.g. A, B, ..., Z, AA, AB...
+            string UniqueParamNameFromIndex(int i)
+            {
+                var alphabetLength = 'Z' - 'A' + 1;
+                var paramName = ((char)('A' + (i % alphabetLength))).ToString();
+
+                while (i / alphabetLength > 0)
+                {
+                    i = (i / alphabetLength) - 1;
+                    var nextCharacter = (char)('A' + (i % alphabetLength));
+                    paramName = nextCharacter + paramName;
+                }
+
+                return paramName;
+            }
```

Comment:
Not strictly necessary, but I'm thinking we can avoid all the intermediate `string` allocations while building the final name, if we just `stackalloc` a buffer with the maximum length and then populate that as necessary. We can rely on the fact that the maximum possible length (even assuming the input is an `uint`) is below 8 characters anyway (since log26(uint.Max) ≈ 6.8), so we can take that as upper bound and then slice at the end as needed. This also produces a `string` with characters in reverse order compared to the original implementation (so we can skip precomputing the final length), but that doesn't matter as the only property we care about is that each value uniquely maps to a text representation, which is preserved here. I'd suggest changing the method as follows, and also making it static:

```suggestion
            static string CreateUniqueParamNameFromIndex(uint i)
            {
                const int alphabetLength = 'Z' - 'A' + 1;

                Span<char> characters = stackalloc char[8];

                characters[0] = (char)('A' + (i % alphabetLength));

                int totalCharacters = 1;

                while ((i /= alphabetLength) > 0)
                {
                    i--;

                    characters[totalCharacters++] = (char)('A' + (i % alphabetLength));
                }

                return characters.Slice(0, totalCharacters).ToString();
            }
```

@michael-hawker @arcadiogarcia If this proposed change is fine by you I can push this myself and also add some XML docs and comments on the code to document the behavior for future reference 😄

Reply:
@Sergio0694 It's definitely fine with me, thanks for the improvements! Feel free to push it yourself if you prefer it 👍

Reply:
Sweet, done in https://github.com/CommunityToolkit/WindowsCommunityToolkit/pull/4112/commits/c5fc961b8a85970fc668cf420b4b8ad0859f03ae! 😄